File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Model/Medico.java
package com.example.BackendCitasMedicas.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Medico {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id_medico;
    private String nombre;
    private String apellido;
    private String correo;
    private String telefono;
    private String especialidad;

    @OneToMany(mappedBy = "medico")
    @JsonIgnore
    private List<Consulta> consultas;

    @OneToMany(mappedBy = "medico")
    @JsonIgnore
    private List<Cita> citas;
}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Model/Medico.kt
package com.example.frontcitasmedicas.Model

data class Medico(
    val id_medico: Long = 0,
    val nombre: String = "",
    val apellido: String = "",
    val correo: String = "",
    val telefono: String = "",
    val especialidad: String = "",
    val consultas: List<Consulta> = emptyList(),
    val citas: List<Cita> = emptyList()
)
